machine_learning/bernoulli_naive_bayes.py

```diff
@@ -0,0 +1,280 @@
+"""
+Naive bayes from scratch on a custom dataset
+
+Naive Bayes Algorithm is a probability based classification technique
+for classifying labelled data. It makes use of Bayes Theorem in order to predict
+the class of the given data. It is called Naive because it makes an assumption that
+the features of the dependent variable are mutually independent of each other.
+Although it is named naive, it is a very efficient model, often used as a baseline
+for text classification and recommender systems.
+
+You can find the detailed notes on this algorithm contributed by me on this notebook: -
+    https://git.io/JTkvO
+"""
+import numpy as np
+import pandas as pd
+
+
+class naive_bayes:
+    def __init__(self, X, y):
+        """
+        This method initializes all the required data fields of the NaiveBayes class
+
+        Input -
+            X: A pandas dataframe consisting of all the dependent variables
+            y: A pandas dataframe consisting of labels
+        """
+        # Initializing the Dependent and Independent Variables
+        self.X = X
+        self.y = y
+
+        # Initializing the column name y. (came in handy for me.
+        # If you do not require it, then you can delete it)
+        self.y_label = y.columns[0]
+
+        # Initializing the variables to store class priors. Initiallt set to None.
+        # They will be the assigned the correct values by executing calculate_prior
+        # p_pos is probability of positive class
+        # p_neg is probability of negative class
+        self.p_pos = None
```

Comment:
```suggestion
        self.probability_pos = None
```
Don't force the reader study to comments.